game/engine/main.py

```
@@ -752,6 +752,110 @@ def handle_bestmove(turn, depth):
+    files = "abcdefgh"
+
+    promo_char = '\0'
+    if is_promotion_move(piece, tr):
+        lower_promo = promo.lower()
+        if lower_promo not in ('q', 'r', 'b', 'n'):
+            lower_promo = 'q'
+        promo_char = lower_promo.upper()
+
+    # 1. Castling
+    if type_ == 'k':
+        if abs(tc - fc) == 2:
```

Ruff (0.15.15)

[warning] Use a single `if` statement instead of nested `if` statements

Combine `if` statements using `and`

(SIM102)